cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -0,0 +1,115 @@
+import abc
+from datetime import datetime
+from typing import Any, ClassVar
+
+from cognite.client.data_classes.data_modeling import PropertyType
+from pydantic import Field, field_validator
+
+from cognite.neat.rules.models._rules.information_rules import InformationMetadata
+
+from .base import BaseMetadata, BaseRules, RoleTypes, SheetEntity, SheetList
+from .domain_rules import DomainMetadata
+
+subclasses = list(PropertyType.__subclasses__())
+_PropertyType_by_name: dict[str, type[PropertyType]] = {}
+for subclass in subclasses:
+    subclasses.extend(subclass.__subclasses__())
+    if abc.ABC in subclass.__bases__:
+        continue
+    try:
+        _PropertyType_by_name[subclass._type.casefold()] = subclass
+    except AttributeError:
+        ...
+del subclasses  # cleanup namespace
+
+
+class DMSArchitectMetadata(BaseMetadata):
+    role: ClassVar[RoleTypes] = RoleTypes.dms_architect
+    space: str
+    external_id: str = Field(alias="externalId")
+    version: str | None = Field(
+        description="Data model version",
+        min_length=1,
+        max_length=43,
+    )
+
+    contributor: str | list[str] = Field(
+        description=(
+            "List of contributors to the data model creation, "
+            "typically information architects are considered as contributors."
+        ),
+    )
+
+    @classmethod
+    def from_information_architect_metadata(
+        cls, metadata: InformationMetadata, space: str | None = None, externalId: str | None = None
+    ):
+        metadata_as_dict = metadata.model_dump()
+        metadata_as_dict["space"] = space or "neat-playground"
+        metadata_as_dict["externalId"] = externalId or "neat_model"
+        return cls(**metadata_as_dict)
+
+    @classmethod
+    def from_domain_expert_metadata(
+        cls,
+        metadata: DomainMetadata,
+        space: str | None = None,
+        external_id: str | None = None,
+        version: str | None = None,
+        contributor: str | list[str] | None = None,
+        created: datetime | None = None,
+        updated: datetime | None = None,
+    ):
+        information = InformationMetadata.from_domain_expert_metadata(
+            metadata, None, None, version, contributor, created, updated
+        ).model_dump()
+
+        return cls.from_information_architect_metadata(information, space, external_id)
+
+
+class DMSProperty(SheetEntity):
```

Comment:
Would it make sense that you inhering from `InformationProperty` and extend it with additional ?
There is already method `is_list` and `is_mandatory`?

Reply:
Composition solution implemented.